### PR TITLE
libobs: Add missing format to format_is_yuv & get_video_format_name

### PR DESCRIPTION
**Pull request by SCG82**
### Description

Insert VIDEO_FORMAT_I422 into switch cases: format_is_yuv and get_video_format_name. Rearrange video formats (in logical order) for consistency: 4:2:0, 4:2:2 (planar, packed), 4:4:4, uncompressed

### Motivation and Context

Fix compiling warning: VIDEO_FORMAT_I422 not included in switch case...

### How Has This Been Tested?

Compilation warning no longer appears.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

File: libobs/media-io/video-io.h [hunk 1]
```diff
@@ -33,11 +33,14 @@ typedef struct video_output video_t;
...
 	VIDEO_FORMAT_NONE,
 
-	/* planar 420 format */
+	/* planar 4:2:0 */
 	VIDEO_FORMAT_I420, /* three-plane */
 	VIDEO_FORMAT_NV12, /* two-plane, luma and packed chroma */
 
-	/* packed 422 formats */
+	/* planar 4:2:2 */
+	VIDEO_FORMAT_I422,
```
Comment (Xaymar) [inline libobs/media-io/video-io.h]: VIDEO_FORMAT_I422 is part of an enum and should not be moved in that enum.

File: libobs/media-io/video-io.h [hunk 2]
```diff
@@ -53,9 +56,6 @@ enum video_format {
 
 	/* more packed uncompressed formats */
 	VIDEO_FORMAT_BGR3,
-
-	/* planar 4:2:2 */
-	VIDEO_FORMAT_I422,
```
Comment (Xaymar) [inline libobs/media-io/video-io.h]: VIDEO_FORMAT_I422 is part of an enum and should not be moved in that enum.

File: libobs/media-io/video-io.h [hunk 3]
```diff
@@ -119,20 +120,22 @@ static inline const char *get_video_format_name(enum video_format format)
...
+	case VIDEO_FORMAT_I444:
+		return "I444";
 	case VIDEO_FORMAT_RGBA:
 		return "RGBA";
 	case VIDEO_FORMAT_BGRA:
 		return "BGRA";
 	case VIDEO_FORMAT_BGRX:
 		return "BGRX";
-	case VIDEO_FORMAT_I444:
-		return "I444";
```
Comment (Xaymar) [inline libobs/media-io/video-io.h]: Was there a reason to move this case?

File: libobs/media-io/video-io.h (hunk 1 above)
Comment (SCG82) [reply libobs/media-io/video-io.h]: fixed

File: libobs/media-io/video-io.h (hunk 3 above)
Comment (SCG82) [reply libobs/media-io/video-io.h]: match the order of the previous switch case, which follows a logical progression: 4:2:0, 4:2:2 (planar, packed), 4:4:4, uncompressed RGB formats

Comment (SCG82) [reply libobs/media-io/video-io.h]: I'll revert that

**Comment by jpark37** [comment]
Looks good to me. Closed #1996.

**Pull request merged**